include/wabt/c-writer.h

```diff
@@ -26,11 +28,20 @@ class Stream;
 
 struct WriteCOptions {
   std::string_view module_name;
+  /* name_to_index takes list of all functions in the module, number of imported
+   * functions, and number of .c outputs as argument, returns a vector where
+   * vector[i] the index of the .c output that funcs[i] goes into. */
+  std::function<std::vector<size_t>(const std::vector<Func*>&,
```

Comment:
I'm realizing that this signature allows the passed-in `name_to_index` function to mutate the functions if it wants. :-/ Is there any clean way to prevent this? We might have to pass a pair of const_iterators and a count of imports, which doesn't seem ideal but I haven't thought of a better way.